File: Examples/test-suite/ruby/template_inherit_runme.rb
#!/usr/bin/env ruby
#
# Put description here
#
# 
# 
# 
#

require 'swig_assert'

require 'template_inherit'

include Template_inherit

a = FooInt.new
b = FooDouble.new
c = BarInt.new
d = BarDouble.new
e = FooUInt.new
f = BarUInt.new

raise ValueError if a.blah() != "Foo"

raise ValueError if b.blah() != "Foo"

raise ValueError if e.blah() != "Foo"

raise ValueError if c.blah() != "Bar"

raise ValueError if d.blah() != "Bar"

raise ValueError if f.blah() != "Bar"

raise ValueError if c.foomethod() != "foomethod"

raise ValueError if d.foomethod() != "foomethod"

raise ValueError if f.foomethod() != "foomethod"

raise ValueError if invoke_blah_int(a) != "Foo"

raise ValueError if invoke_blah_int(c) != "Bar"

raise ValueError if invoke_blah_double(b) != "Foo"

raise ValueError if invoke_blah_double(d) != "Bar"

raise ValueError if invoke_blah_uint(e) != "Foo"

raise ValueError if invoke_blah_uint(f) != "Bar"

File: Examples/test-suite/javascript/template_inherit_runme.js
var template_inherit = require("template_inherit");
a = new template_inherit.FooInt();
b = new template_inherit.FooDouble();
c = new template_inherit.BarInt();
d = new template_inherit.BarDouble();
e = new template_inherit.FooUInt();
f = new template_inherit.BarUInt();

if (a.blah() != "Foo") {
    throw new Error;
}

if (b.blah() != "Foo") {
    throw new Error;
}

if (e.blah() != "Foo") {
    throw new Error;
}

if (c.blah() != "Bar") {
    throw new Error;
}

if (d.blah() != "Bar") {
    throw new Error;
}

if (f.blah() != "Bar") {
    throw new Error;
}

if (c.foomethod() != "foomethod") {
    throw new Error;
}

if (d.foomethod() != "foomethod") {
    throw new Error;
}

if (f.foomethod() != "foomethod") {
    throw new Error;
}

if (template_inherit.invoke_blah_int(a) != "Foo") {
    throw new Error;
}

if (template_inherit.invoke_blah_int(c) != "Bar") {
    throw new Error;
}

if (template_inherit.invoke_blah_double(b) != "Foo") {
    throw new Error;
}

if (template_inherit.invoke_blah_double(d) != "Bar") {
    throw new Error;
}

if (template_inherit.invoke_blah_uint(e) != "Foo") {
    throw new Error;
}

if (template_inherit.invoke_blah_uint(f) != "Bar") {
    throw new Error;
}
